types/p-event/index.d.ts

```diff
@@ -0,0 +1,34 @@
+// Type definitions for p-event 1.3
+// Project: https://github.com/sindresorhus/p-event#readme
+// Definitions by: BendingBender <https://github.com/BendingBender>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+export = pEvent;
+
+declare function pEvent<T, TRest = T>(emitter: pEvent.Emitter<T, TRest>, event: string | symbol, options: MultiArgsOptions): Promise<Array<T | TRest>>;
+declare function pEvent<T>(emitter: pEvent.Emitter<T>, event: string | symbol, options?: pEvent.Options): Promise<T>;
+
+declare namespace pEvent {
+    interface Emitter<T = any, TRest = T> {
+        on?: AddRmListenerFn<T, TRest>;
```

Comment:
From looking at this https://www.npmjs.com/package/p-event#emitter shouldnt this Emitter be of type 
```TypeScript
type Emitter<T, TRest = T> {
on: AddRmListenerFn<T, TRest>;
off: AddRmListenerFn<T, TRest>;
} | {
addListener?: AddRmListenerFn<T, TRest>;
removeListener?: AddRmListenerFn<T, TRest>;
} | {
addEventListener?: AddRmListenerFn<T, TRest>;
removeListener?: AddRmListenerFn<T, TRest>;
}
```